src/main/ruby/core/truffle/kernel_operations.rb

```diff
@@ -14,5 +14,90 @@ def self.define_hooked_variable(name, getter, setter, defined = proc { 'global-v
       defined = Truffle::Graal.always_split(defined) if defined.arity == 1
       define_hooked_variable_with_is_defined(name, getter, setter, defined)
     end
+
+    def self.define_read_only_global(name, getter)
+      setter = -> _ { raise NameError, "#{name} is a read-only variable." }
+      define_hooked_variable(name, getter, setter)
+    end
+
+    LOAD_PATH = []
+    LOADED_FEATURES = []
   end
 end
+
+Truffle::KernelOperations.define_read_only_global(:$LOAD_PATH, -> { Truffle::KernelOperations::LOAD_PATH })
+Truffle::KernelOperations.define_read_only_global(:$LOADED_FEATURES, -> { Truffle::KernelOperations::LOADED_FEATURES })
```

Comment:
If these variables and below were define inside `module Truffle::KernelOperations` it would make repeating that prefix unnecessary.